### PR TITLE
DEP: special: Remove `special.btdtr` and `special.btdtri`

### LINKED ISSUE #21759: DEP: remove `special.btdtr` and `special.btdtri`
### Describe your issue.

Both functions have been announced to be removed in 1.14:

https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.btdtri.html
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.btdtr.html

These were originally deprecated in https://github.com/scipy/scipy/pull/17697/commits/40ebe8dac578e03e5ccfd466a9ed4a205496d79d from #17697

### PR DESCRIPTION
**Pull request by IamPhytan**
#### Reference issue

Closes #21759.

#### What does this implement/fix?

Remove references to `special.btdtr` and `special.btdtri`. Scheduled to be removed in 1.14.0.

### CODE REVIEW

File: scipy/special/__init__.py [hunk 1]
```diff
@@ -880,25 +878,6 @@ def _load_libsf_error_state():
 _depr_msg = ('\nThis function was deprecated in SciPy 1.12.0, and will be '
              'removed in SciPy 1.14.0.  Use scipy.special.{} instead.')
```
Comment (dschmitz89) [inline scipy/special/__init__.py:879]:
```suggestion
```

If this is not used anywhere else, let's remove it while we are at it.

**Review by dschmitz89 — CHANGES_REQUESTED** [review]
Thanks for the PR @IamPhytan . I think the biggest obstacle is the CircleCI timeout which is unrelated to your changes. Let's hope that the next CI run does not run in the same issue.

**Comment by IamPhytan** [comment]
Perfect!
I commited your suggested change, @dschmitz89

**Review by j-bowhay — APPROVED** [review]
LGTM thanks!

**Pull request merged**